src/lib.rs

```diff
@@ -498,9 +498,32 @@ macro_rules! ImplDekuTraits {
     };
 }
 
-impl<T: DekuRead<Ctx>, Ctx: Copy> DekuRead<(Count, Ctx)> for Vec<T> {
-    /// Read the specified number of `T`s from input.
-    /// * `count` - the number of `T`s you want to read.
+fn read_vec_with_predicate<T: DekuRead<Ctx>, Ctx: Copy, Predicate: FnMut(usize, &T) -> bool>(
+    input: &BitSlice<Msb0, u8>,
+    vec: Vec<T>,
```

Comment:
nit: possibly change to `&mut Vec<T>`

Reply:
Thinking about it, perhaps it's actually nicer to just take an optional capacity instead of the vec directly, and construct the vec inside this method? That feels slightly less weird to me than passing in the vec did, at least